build.zig

```diff
@@ -60,16 +63,92 @@ pub fn build(b: *std.Build) void {
 
             lib.addIncludePath(b.path(cache_include));
         },
-        else => {
-            const config_header = b.addConfigHeader(.{
-                .style = .{ .cmake = b.path("include/SDL_config.h.cmake") },
-                .include_path = "SDL2/SDL_config.h",
-            }, .{});
-            lib.addConfigHeader(config_header);
-            lib.installConfigHeader(config_header);
-        },
+        else => { },
+    }
+
+    if (use_pregenerated_config) {
+        lib.addCSourceFiles(.{ .files = render_driver_sw.src_files });
+        lib.addIncludePath(b.path("include"));
+        lib.installHeadersDirectory(b.path("include"), "SDL2", .{});
+    } else {
+        var files = std.ArrayList([]const u8).init(b.allocator);
+        defer files.deinit();
+
+        // copy all headers except the pregenerated SDL_config.h and SDL_revision.h
+        // so we can use our configured headers instead (SDL's CMakeLists.txt does
+        // the same thing)
+        const write_files = b.addWriteFiles();
+        inline for (static_headers) |header| {
+            _ = write_files.addCopyFile(b.path("include/" ++ header), header);
+        }
+        lib.addIncludePath(write_files.getDirectory());
```

Comment:
it seems sus to do this rather than having 2 include paths, but I guess if that's what SDL does with their build then it's fine.

Reply:
that sounds like a way better idea, gonna try this